pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
             cat(schemaString)
           })
 
+#' Get schema object
+#' 
+#' Returns the schema of this DataFrame as a list of named lists. Each named
```

Comment:
@shivaram I'm interested in your take on this as well. I know that PySpark handles all of the DataTypes with a series of classes and superclasses, which, as @davies points out, would be a lot harder to replicate in S4. I just assumed we were going to have to extend the backend to handle all of these going both ways, but if there's an easier way then great. I'm just not sure I follow what the wrapper function would look like in this case.